### ISSUE TITLE
t/ffi_c_util.t fails with -Duselongdouble

### ISSUE THREAD

**Issue by eserte** [Perl5-FFI/FFI-C#24]
If perl was compiled with -Duselongdouble, then the test suite fails:
```
        # Failed test 'c_to_perl'
        # at t/ffi_c_util.t line 122.
        # +-------------+---------------------+----+-------+
        # | PATH        | GOT                 | OP | CHECK |
        # +-------------+---------------------+----+-------+
        # | {y}[1]{baz} | 8.80000000000000071 | eq | 8.8   |
        # +-------------+---------------------+----+-------+
    # Failed test 'generated classes'
    # at t/ffi_c_util.t line 135.
        # Failed test 'c_to_perl'
        # at t/ffi_c_util.t line 180.
        # +-------------+---------------------+----+-------+
        # | PATH        | GOT                 | OP | CHECK |
        # +-------------+---------------------+----+-------+
        # | {y}[1]{baz} | 8.80000000000000071 | eq | 8.8   |
        # +-------------+---------------------+----+-------+
    # Failed test 'ungenerated types'
    # at t/ffi_c_util.t line 194.

# Failed test 'perl_to_c / c_to_perl'
# at t/ffi_c_util.t line 196.
t/ffi_c_util.t ....... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/2 subtests 
```

**Comment by plicease** [comment]
Both longdouble and quadmath are affected
https://travis-ci.org/github/Perl5-FFI/FFI-C/builds/682581848
It's a bad test that is relying on roundtrip of floating points when you can't quite count on when converting long double and back.
#25

**Comment by plicease** [comment]
Should be fixed in `0.02`.  I've also added longdouble and quadmath to CI so should catch any regressions here.
